### PR TITLE
Fix reshape for output and ensemble

### PR DESCRIPTION
**Pull request by GuanLuo**

### CODE REVIEW

File: src/core/infer_response.cc [hunk 1]
```diff
@@ -73,6 +73,12 @@ InferenceResponse::AddOutput(
 
   LOG_VERBOSE(1) << "add response output: " << outputs_.back();
 
+  if (backend_ != nullptr) {
```
Comment (deadeyegoodwin) [inline src/core/infer_response.cc:76]: backend_raw_ is guaranteed to be non-null so you should use that.

File: src/core/infer_response.h [hunk 2]
```diff
@@ -137,6 +137,9 @@ class InferenceResponse {
     // The shape of the output tensor.
     const std::vector<int64_t>& Shape() const { return shape_; }
 
+    // Reshape the output tensor if specified in output config.
+    void Reshape(const ModelOutput* output_config);
```
Comment (deadeyegoodwin) [inline src/core/infer_response.h:141]: This can be private

File: src/core/infer_response.cc (hunk 1 above)
Comment (GuanLuo) [reply src/core/infer_response.cc:76]: InferenceResponse class only has `backend_`

File: src/core/infer_response.h (hunk 2 above)
Comment (GuanLuo) [reply src/core/infer_response.h:141]: `Rehape()` is member function of `InferenceResponse::Output` and called in `InferenceResponse`

**Pull request merged**